softvis3d-frontend/src/classes/TreeElement.ts

```diff
@@ -0,0 +1,56 @@
+export class TreeElement {
+
+    public readonly id: string;
+    public readonly key: string;
+    public measures: MeasureList;
+
+    public name: string;
+    public readonly path: string;
+
+    public parent?: TreeElement;
+    public readonly children: TreeElement[];
+
+    public isFile: boolean;
+
+    public constructor(id: string, key: string, measures: MeasureList, name: string, path: string, isFile: boolean,
+                       parent?: TreeElement) {
+        this.id = id;
+        this.key = key;
+        this.measures = measures;
+        this.name = name;
+        this.path = path;
+        this.isFile = isFile;
+        this.parent = parent;
+        this.children = [];
+    }
+
+    /**
+     * Will be called with the path of the components sorted.
+     */
+    public addAsChild(element: TreeElement, isDescending: boolean = false) {
+        let children = this.children;
+        if (isDescending) {
+            children = children.reverse();
```

Comment:
I see. Since the algorithm does nothing close to an ordered insert this behavior would ascribe to the input data (SonarQube) though and is not universally applicable, right?